Marlin/src/HAL/HAL_STM32/HAL.h

```diff
@@ -132,7 +132,7 @@
 // Types
 // --------------------------------------------------------------------------
 
-typedef int8_t pin_t;
+typedef int16_t pin_t;
```

Comment:
Do you suppose this is okay for all STM32 platforms, or would some be okay with 8 bits? Is STM32 one of those MCUs where 16 bit integers are more optimal, so it's no difference to use this for all of them?